waflib/Tools/fc_scan.py

```diff
@@ -43,22 +47,31 @@ def find_deps(self, node):
 		:return: lists representing the includes, the modules used, and the modules created by a fortran file
 		:rtype: tuple of list of strings
 		"""
-		txt = node.read()
-		incs = []
-		uses = []
-		mods = []
-		for line in txt.splitlines():
-			# line by line regexp search? optimize?
-			m = re_inc.search(line)
-			if m:
-				incs.append(m.group(1))
-			m = re_use.search(line)
-			if m:
-				uses.append(m.group(1))
-			m = re_mod.search(line)
-			if m:
-				mods.append(m.group(1))
-		return (incs, uses, mods)
+		try:
+			cache = node.ctx.cache_fc_scan_deps
+		except AttributeError:
+			cache = node.ctx.cache_fc_scan_deps = Utils.lru_cache(DEPS_CACHE_SIZE)
+		try:
+			return cache[node]
+		except KeyError:
+			txt = node.read()
+			incs = []
+			uses = []
+			mods = []
+			for line in txt.splitlines():
+				# line by line regexp search? optimize?
+				m = re_inc.search(line)
+				if m:
+					incs.append(m.group(1))
+				m = re_use.search(line)
+				if m:
+					uses.append(m.group(1))
+				m = re_mod.search(line)
+				if m:
+					mods.append(m.group(1))
+			ret = (incs, uses, mods)
```

Comment:
This isn't a huge issue by any means, but it might be clearer to use an object here.